### PR TITLE
feat: Get bnb balance hook

### PR DESCRIPTION
**Pull request by hachiojidev**
A hook for getting a wallet's BNB balance

### CODE REVIEW

File: src/hooks/useTokenBalance.ts [hunk 1]
```diff
@@ -61,4 +63,24 @@ export const useBurnedBalance = (tokenAddress: string) => {
...
+      const walletBalance = await web3.eth.getBalance(account)
+      setBalance(new BigNumber(walletBalance))
+    }
+
+    if (account) {
+      fetchBalance()
+    }
+  }, [account, lastUpdated, setBalance])
+
+  return { balance, refresh: setLastUpdated }
```
Comment (hachiojidev) [inline src/hooks/useTokenBalance.ts:83]: `refresh` is so the consumer of the hook can re-fetch the balance at their leisure

File: src/hooks/useTokenBalance.ts [hunk 2]
```diff
@@ -61,4 +63,24 @@ export const useBurnedBalance = (tokenAddress: string) => {
...
+export const useGetBnbBalance = () => {
+  const [balance, setBalance] = useState(new BigNumber(0))
+  const { account } = useWeb3React()
+  const { lastUpdated, setLastUpdated } = useLastUpdated()
+
+  useEffect(() => {
+    const fetchBalance = async () => {
+      const web3 = getWeb3NoAccount()
+      const walletBalance = await web3.eth.getBalance(account)
+      setBalance(new BigNumber(walletBalance))
```
Comment (Chef-Cheems) [inline src/hooks/useTokenBalance.ts:75]: Don't you need `web3.utils.fromWei(walletBalance, "ether")` here? Or it's supposed to be in Wei?

File: src/hooks/useTokenBalance.ts (hunk 1 above)
Comment (ChefHutch) [reply src/hooks/useTokenBalance.ts:83]: This is neat. I actually hadn't seen you add that `useLastUpdated` hook - it should be pretty useful! I found myself toggling a state boolean at points in the trading comp to effectively do the same thing.

File: src/hooks/useTokenBalance.ts (hunk 2 above)
Comment (hachiojidev) [reply src/hooks/useTokenBalance.ts:75]: I wasn't aware of that utility but I think most of our balances return as Wei.

**Review by ChefHutch — APPROVED** [review]
👍

File: src/hooks/useTokenBalance.ts [hunk 3]
```diff
@@ -61,4 +63,24 @@ export const useBurnedBalance = (tokenAddress: string) => {
...
 }
 
+export const useGetBnbBalance = () => {
+  const [balance, setBalance] = useState(new BigNumber(0))
+  const { account } = useWeb3React()
+  const { lastUpdated, setLastUpdated } = useLastUpdated()
+
+  useEffect(() => {
+    const fetchBalance = async () => {
+      const web3 = getWeb3NoAccount()
```
Comment (Chef-Cheems) [inline src/hooks/useTokenBalance.ts:73]: Shouldn't this be `const web3 = useWeb3()`?

Comment (hachiojidev) [reply src/hooks/useTokenBalance.ts:73]: Yeah...why does this work? :)

Comment (hachiojidev) [reply src/hooks/useTokenBalance.ts]: @Chef-Cheems updated

Comment (Chef-Cheems) [reply src/hooks/useTokenBalance.ts]: @hachiojidev you forgot to remove getWeb3NoAccount import :)
Yeah regarding how it works.... I dunno, I guess Web3NoAccount is still valid Web3 that can query account balances via address, so in this case it doesn't really mater? 🤔 
I haven't looked that deep into web3 setup in PCS yet but some stuff seems a bit weird for me, there might be some stuff to optimize or at least to document 😅

**Review by Chef-Chungus — APPROVED** [review]

**Pull request merged**